scripts/tuple_tree_generator/tuple_tree_generator/templates/tuple_tree_gen.py.tpl

```diff
@@ -95,6 +101,20 @@ class 'struct.name'(
 
 ## endfor ##
 
+## for struct in structs ##
+## if struct.abstract -##
+'struct.name'._children = {
+        ##- for child in struct.children -##
+        "'-child.name'": '-child.name' ##- if not loop.last -## , ##- endif -##
+        ##- endfor -##
+    }
```

Comment:
Mmmh, this out-of-line definition doesn't sound right. Can we push this in the body of the `class` definition?

Reply:
No, because there is a cyclic dependency:
* The base class needs to be defined for the children to subclass it
* The children would need to be before the base class to have a child list in the base class

This is (AFAIK) the most straightforward way of solving this.

Reply:
Sad, Python really needs forward declarations `:P`